tiledb/api/src/string.rs

```diff
@@ -34,15 +34,11 @@ impl TDBString {
     }
 
     pub fn to_string(&self) -> TileDBResult<String> {
-        let mut c_str: *const i8 = out_ptr!();
+        let mut c_str = out_ptr!();
```

Comment:
Since these are non-obvious, I'll note that the API uses `c_char` in its signature. Unfortunately, `c_char`'s signedness isn't defined and I happened to be the first to find a platform where its `u8`. Rather than cast, I've just removed the type and rely on inference now.